sql/src/main/java/org/apache/druid/sql/SqlStatementFactory.java

```diff
@@ -23,11 +23,30 @@
 
 import javax.servlet.http.HttpServletRequest;
 
-public interface SqlStatementFactory
+public class SqlStatementFactory
 {
-  HttpStatement httpStatement(SqlQuery sqlQuery, HttpServletRequest req);
+  private final SqlToolbox lifecycleToolbox;
 
-  DirectStatement directStatement(SqlQueryPlus sqlRequest);
+  public SqlStatementFactory(SqlToolbox lifecycleToolbox)
```

Comment:
Perhaps add a comment to point the reader to the `SqlModule` and `SqlStatementFactoryModule` to explain how these instances are created. It is, unfortunately, not intuitively obvious on the first read how things are now wired up.

Reply:
There is magic in DI platforms, to a certain degree, you want to know the magic to know how things are wired up.

Generally speaking, there is one rule with using Guice that we should be following (I noticed that we have drifted from the rules in some of the code) to help know how things are wired up.  Never, ever, ever, build on or expect implicit bindings (in the code I was looking at, I noticed a number of implicit bindings that were happening, I made explicit ones for everything that I found so far).  We should always have an explicit binding in a module somewhere for everything that we expect to be bound.  The reason for this is that without an explicit binding, it's impossible to do a usages search and figure out how things are setup.  With an explicit binding, you can do a usages search on the class and then find the usages "as a `.class`" which tells you which Module it is bound in.  Provider methods are kinda a gray area here and I generally try to keep away from them if possible.  In this case, without revisiting the object structure in general, either a provider method or an actual provider would be required.  For Provider methods, the `.class` usages don't show up, but if you also look for usages as a return value from a method, you can find the Provider methods as well.  Additionally, usages of the constructor will show up here.

TBH, I would also just add the javadoc, but I'm scared of another multi-hour CI cycle blocking the PR.  Maybe I'll do the javadoc and merge anyway.